core/src/layers/mod.rs

```diff
@@ -18,37 +18,53 @@
 //! `Layer` is the mechanism to intercept operations.
 
 mod concurrent_limit;
+
 pub use concurrent_limit::ConcurrentLimitLayer;
 
 mod immutable_index;
+
 pub use immutable_index::ImmutableIndexLayer;
 
 mod logging;
+
 pub use logging::LoggingLayer;
 
 #[cfg(feature = "layers-chaos")]
 mod chaos;
+
 #[cfg(feature = "layers-chaos")]
 pub use chaos::ChaosLayer;
 
 #[cfg(feature = "layers-metrics")]
 mod metrics;
+
 #[cfg(feature = "layers-metrics")]
 pub use self::metrics::MetricsLayer;
 
 mod retry;
+
 pub use self::retry::RetryLayer;
 
 #[cfg(feature = "layers-tracing")]
 mod tracing;
+
 #[cfg(feature = "layers-tracing")]
 pub use self::tracing::TracingLayer;
 
 mod type_eraser;
+
 pub(crate) use type_eraser::TypeEraseLayer;
 
 mod error_context;
+
 pub(crate) use error_context::ErrorContextLayer;
 
 mod complete;
+
 pub(crate) use complete::CompleteLayer;
+
+mod simulation;
+
+pub use simulation::MadsimLayer;
+#[cfg(madsim)]
```

Comment:
I believe opendal is not a right place for adding `cfg`.

- OpenDAL adds a new feautre to control whether we provide madsim layer.
- Users can use the way they prefer to control the logic.

Reply:
I'm sorry that `cfg` cannot be removed totally, as some core mods of madsim can only be compiled with ` RUSTFLAGS="--cfg madsim"', but I have changed madsim layer into a feature.